packages/ui-extensions-react/src/surfaces/point-of-sale/render.tsx

```diff
@@ -39,22 +39,21 @@ export function reactExtension<Target extends RenderExtensionTarget>(
   return extension<'pos.home.tile.render'>(target as any, async (root, api) => {
     const element = await render(api as ApiForRenderExtension<Target>);
 
-    await new Promise<void>((resolve, reject) => {
+    return new Promise<() => void>((resolve, reject) => {
       try {
-        remoteRender(
+        const remoteRoot = createRoot(root);
```

Comment:
I'm not exactly sure what else would be required in this file, but this seems to work properly. Also it would be nice to know if this change is desired elsewhere. @vividviolet you mentioned we could just share this function implementation instead of having it be duplicated everywhere.